matrix-e2e.c

```diff
@@ -39,6 +39,106 @@ struct _MatrixE2EData {
     sqlite3 *db;
 };
 
+#define PURPLE_CONV_E2E_STATE "e2e"
+
+/* Hung off the Purple conversation with the PURPLE_CONV_E2E_STATE */
+typedef struct _MatrixE2ERoomData {
+    /* Mapping from _MatrixHashKeyInBoundMegOlm to OlmInboundGroupSession */
+    GHashTable *megolm_sessions_inbound;
+} MatrixE2ERoomData;
+
+typedef struct _MatrixHashKeyInBoundMegOlm {
```

Comment:
I'd want to sanity check the device/sender id's anyway - so it seemed
to make sense to add them in the hash.